library/common/event/provisional_dispatcher.cc

```diff
@@ -49,5 +49,21 @@ void ProvisionalDispatcher::deferredDelete(DeferredDeletablePtr&& to_delete) {
   event_dispatcher_->deferredDelete(std::move(to_delete));
 }
 
+void ProvisionalDispatcher::pushTrackedObject(const ScopeTrackedObject* object) {
+  if (event_dispatcher_) {
```

Comment:
This access is not inherently threadsafe, though it will likely only be called from a valid context. Still, we might want to have a safety check in these calls (like we do with deferredDelete).

Reply:
currently scopes are only created in the threading context of the event dispacther. I removed all if statements and put in asserts like in deferred delete